### ISSUE TITLE
MemoizedInstanceVariableName cop suggests syntactically invalid instance variable for predicate methods.

### ISSUE THREAD

**Issue by cabello** [bbatsov/rubocop#5670]
Continuation of https://github.com/bbatsov/rubocop/issues/5648

### Expected behavior
It should **not** suggest replacing `@full_name` with `@full_name!`

### Actual behavior

```
app/models/client.rb:93:5: C: Naming/MemoizedInstanceVariableName: Memoized variable @full_name does not match method name full_name!. Use @full_name! instead.
```

Rubocop wants me to change the instance variable to `@full_name!`, which is not valid.

### Steps to reproduce the problem
```ruby
def full_name!
  @full_name ||= 'fetch full name'
end
```

### RuboCop version
0.53.0
